src/main/java/org/freeswitch/esl/client/internal/AbstractEslClientHandler.java

```diff
@@ -225,23 +226,51 @@ protected void handleEslMessage(ChannelHandlerContext ctx, EslMessage message) {
 	 * @param command
 	 * @return Job-UUID as a string
 	 */
-	public CompletableFuture<EslEvent> sendBackgroundApiCommand(Channel channel, final String command) {
-
+	public CompletableFuture<String> sendBackgroundApiJobCommand(Channel channel, final String command) {
 		return sendApiSingleLineCommand(channel, command)
 				.thenComposeAsync(result -> {
 					if (result.hasHeader(Name.JOB_UUID)) {
 						final String jobId = result.getHeaderValue(Name.JOB_UUID);
-						final CompletableFuture<EslEvent> resultFuture = new CompletableFuture<>();
-						backgroundJobs.put(jobId, resultFuture);
-						return resultFuture;
+						return CompletableFuture.completedFuture(jobId);
 					} else {
-						final CompletableFuture<EslEvent> resultFuture = new CompletableFuture<>();
+						final CompletableFuture<String> resultFuture = new CompletableFuture<>();
 						resultFuture.completeExceptionally(new IllegalStateException("Missing Job-UUID header in bgapi response"));
 						return resultFuture;
 					}
 				}, backgroundJobExecutor);
 	}
 
+
+	/**
+	 * Returns the EslEvent of that the response event will have.
+	 *
+	 * @param channel
+	 * @param command
+	 * @return EslEvent
+	 */
+	public CompletableFuture<EslEvent> sendBackgroundApiCommand(Channel channel, final String command) {
+		return sendBackgroundApiJobCommand(channel, command).thenComposeAsync(jobId -> {
+			final CompletableFuture<EslEvent> resultFuture = new CompletableFuture<>();
+			backgroundJobs.put(jobId, resultFuture);

```

Comment:
I think if sendBackgroundApiJobCommand **completeExceptionally**, jobId will null

Reply:
I think composeAsync just runs when there is no exception 

Ref : https://stackoverflow.com/questions/47044984/using-java-completablyfuture-exceptionally-to-optionally-continue-through-exce